### PR TITLE
feat: Enhanced the Contact us Page 

### LINKED ISSUE #663: [FEATURE] Want to enhance the contact us page for the websire
### Is your feature request related to a problem? Please describe.

I want to enhance the contact us page for the website 
I will improve the designing of the web page 

### Describe the solution you'd like.

I will make the contact page more attractive 

### Screenshots

image

I WILL CONVERT IT TO THIS 

image

### Additional information

_No response_

### Do you want to work on this issue?

Yes

### PR DESCRIPTION
**Pull request by Ayushmaanagarwal1211**
## Fixes issue

Fixes #663

## Changes proposed

I have enhanced the contact us page for the website as shown in the image 

## Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
image

## Note to reviewers

Please review this and let me know if any changes 

## Summary by CodeRabbit

- **New Features**
	- Redesigned contact form layout for improved responsiveness and usability.
	- Updated placeholder text in the message textarea to "How Can I Help You?".
	- Added information about Terms of Service and Privacy Policy at the bottom of the form.

- **Bug Fixes**
	- Improved email input validation with an updated regex pattern.
	- Enhanced phone number input to accept a broader range of formats, including international numbers.
	- Enforced a minimum character requirement for the name input.

- **Style**
	- Enhanced styling for input fields and buttons, including rounded corners, shadow effects, and hover effects.

### CODE REVIEW

**Comment by SaptarshiSarkar12** [comment]
@Ayushmaanagarwal1211 Awesome work 👏.
Some questions from my side: 
- You marked that there is a need for change in documentation but I don't think there is any. Let me know your views. 🙂
- can you please make the changes listed in the last Coderabbit AI's review comment (under the collapsed tag) except the last review request? Let me know if you think any of those is not required.

**Comment by Ayushmaanagarwal1211** [comment]
@SaptarshiSarkar12 oh so sorry i just checked it by mistake 

And actually i don't know too much about coderabbit but i did change in the regex pattern of email field for fix this 
Can you please explain me little bit more so that i can make changes according it

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@Ayushmaanagarwal1211 okay. So, here's what changes are requested by Coderabbit AI 👇

`13-15`: **Consider removing unnecessary empty lines.**

These empty lines don't separate logical sections of the code and can be removed to improve code conciseness.

Apply this change:

```diff
-     
-      
-
```

---

`21-28`: **Good form setup and header, consider adding aria-label.**

The form container and header are well-structured and visually appealing. The use of FormSubmit for handling submissions is a good choice.

For improved accessibility, consider adding an `aria-label` to the form element:

```diff
 <form
   className="md:w-3/5   bg-white rounded-2xl shadow-lg shadow-gray-500 p-14" 
   method="POST"
   action="https://formsubmit.co/e94b201f9a607081e9f9f8ee09ff5e25"
+  aria-label="Contact Us Form"
 >
```

---

`30-47`: **Well-styled inputs with proper validation, consider adding name pattern.**

The input fields for name and email are well-styled and include proper validation. The email pattern has been correctly updated.

For consistency, consider adding a pattern attribute to the name input as well:

```diff
 <input
   className= "border max-md:w-full rounded-full max-w-[400px] p-2 w-1/2 bg-white  focus:outline-blue-600 text-blue-600 outline:none caret-blue-600 md:w-6/12 sm:w-1/2  h-12  col-span-2"
   type="text"
   name="name"
   autoFocus={true}
   inputMode="text"
   required
   placeholder="Name"
+  pattern="^[A-Za-z\s]{1,}$"
 />
```

This pattern will ensure that only alphabets and spaces are allowed in the name field.

---

Line range hint `49-66`: **Good textarea update, consider revising phone number pattern.**

The textarea placeholder update to "How Can I Help You?" improves user understanding. However, the phone number pattern might be too restrictive for international formats.

Consider using a more flexible pattern for phone numbers:

```diff
 <input
   className="border rounded-full p-2 w-full my-2 bg-white  focus:outline-blue-600 outline:none caret-blue-600 mt-6 md:mr-auto md:ml-1 justify-start  h-12   md:col-span-1 col-span-2 text-blue-600"
   minLength={6}
-  pattern="/^([+]?\d{1,2}[-\s]?|)\d{3}[-\s]?\d{3}[-\s]?\d{4}$/"
+  pattern="^[+]?[(]?[0-9]{3}[)]?[-\s.]?[0-9]{3}[-\s.]?[0-9]{4,6}$"
   maxLength={26}
   inputMode="tel"
   type="tel"
   name="phone"
   required
   placeholder="Phone Number"
 />
```

This pattern allows for more variations in phone number formats while still providing basic validation

**Comment by Ayushmaanagarwal1211** [comment]
@SaptarshiSarkar12 thanks for explaining i am doing please wait

**Comment by Ayushmaanagarwal1211** [comment]
@SaptarshiSarkar12 can you please check again

File: Website/app/contact/page.js [hunk 1]
```diff
@@ -10,38 +10,44 @@ export default function contact() {
...
           <input
-            className="bg-white rounded-md focus:outline-blue-600 text-blue-600 outline:none caret-blue-600 md:w-6/12 sm:w-1/2 h-14 p-3 col-span-2"
+            className= "border max-md:w-full rounded-full max-w-[400px] p-2 w-1/2 bg-white  focus:outline-blue-600 text-blue-600 outline:none caret-blue-600 md:w-6/12 sm:w-1/2  h-12  col-span-2"
             type="text"
             name="name"
             autoFocus={true}
             inputMode="text"
             required
             placeholder="Name"
+            pattern="^[A-Za-z\s]{1,}$"
```
Comment (SaptarshiSarkar12) [inline Website/app/contact/page.js:35]: @Ayushmaanagarwal1211 Please add a statement about the allowed characters in all the areas where we have added pattern. You can check this for reference - https://stackoverflow.com/a/19123055

Comment (SaptarshiSarkar12) [inline Website/app/contact/page.js:35]: Allow `'` and `-` for some specific types of names.
```suggestion
            pattern="^[A-Za-z\s'-]{1,}$"
```

File: Website/app/contact/page.js [hunk 2]
```diff
@@ -10,38 +10,44 @@ export default function contact() {
...
             required
             placeholder="Email"
           />
+          </div>
           <input
-            className="bg-white rounded-md focus:outline-blue-600 outline:none caret-blue-600 mt-10 md:mr-auto md:ml-1 md:w-1/2 justify-start  h-14 p-2  md:col-span-1 col-span-2 text-blue-600"
+            className="border rounded-full p-2 w-full my-2 bg-white  focus:outline-blue-600 outline:none caret-blue-600 mt-6 md:mr-auto md:ml-1 justify-start  h-12   md:col-span-1 col-span-2 text-blue-600"
             minLength={6}
-            pattern="/^([+]?\d{1,2}[-\s]?|)\d{3}[-\s]?\d{3}[-\s]?\d{4}$/"
+            pattern="^[+]?[(]?[0-9]{3}[)]?[-\s.]?[0-9]{3}[-\s.]?[0-9]{4,6}$"
```
Comment (SaptarshiSarkar12) [inline Website/app/contact/page.js:50]: Do you think this change would be good? Try the possibilities and tell me.
```suggestion
            pattern="^(\+\d{1,3}[- ]?)?\d{10}$"
```

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@Ayushmaanagarwal1211 Please make these changes.

File: Website/app/contact/page.js (hunk 2 above)
Comment (Ayushmaanagarwal1211) [reply Website/app/contact/page.js:50]: yeah ^(\+\d{1,3}[- ]?)?\d{10}$" this would be better according to me

**Comment by Ayushmaanagarwal1211** [comment]
@SaptarshiSarkar12 i did these changes  i have added a statement and updated the regex please check if need some changes in statement please let me know 
image

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@Ayushmaanagarwal1211 By statements, I didn't mean adding texts beneath the text boxes. As per that StackOverflow answer, they are using `title` attribute in input section to get this kind of output 👇 
Screenshot_20241002-151422
Make this change, rest all are fine.

File: Website/app/contact/page.js [hunk 3]
```diff
@@ -10,57 +10,70 @@ export default function contact() {
...
+  <p>We&apos;d love to hear from you!</p>
+  <div className="flex justify-between gap-2 mt-6  max-md:flex-col">
+      <input
+        type="text"
+        autoFocus={true}
+        inputMode="text"
+        required
+        placeholder="Name"
+        pattern="^[a-zA-Z_]{5,}$"
+        title="Please enter at least 5 characters and a - z , A - Z and _ are allowed"
```
Comment (SaptarshiSarkar12) [inline Website/app/contact/page.js:31]:
```suggestion
        title="Please enter at least 5 characters. Only lowercase, uppercase and underscore are allowed"
```

File: Website/app/contact/page.js [hunk 4]
```diff
@@ -10,57 +10,70 @@ export default function contact() {
...
+        className="h-[50px] border pl-3 min-w-[200px] max-md:w-[95%] w-[40%]  border-gray-300 rounded-3xl"
+      />
+      <input
+        className="h-[50px] border pl-3  min-w-[200px] max-md:w-[95%]  w-[40%] max-w-[500px] border-gray-300 rounded-3xl"
+        type="email"
+        required
+        pattern="^[A-Za-z\s'-]{5,}$"
```
Comment (SaptarshiSarkar12) [inline Website/app/contact/page.js:38]: @Ayushmaanagarwal1211 Why do you think this pattern change is required?

File: Website/app/contact/page.js [hunk 5]
```diff
@@ -10,57 +10,70 @@ export default function contact() {
...
+        id="email"
+        autoFocus
```
Comment (SaptarshiSarkar12) [inline Website/app/contact/page.js:40]:
```suggestion
```

**Review by SaptarshiSarkar12 — CHANGES_REQUESTED** [review]
@Ayushmaanagarwal1211 Make this changes.

**Comment by SaptarshiSarkar12** [comment]
@Ayushmaanagarwal1211 Please make these changes 
- https://github.com/SaptarshiSarkar12/Drifty/pull/664#discussion_r1784345022
- https://github.com/SaptarshiSarkar12/Drifty/pull/664#discussion_r1784348579
- https://github.com/SaptarshiSarkar12/Drifty/pull/664#discussion_r1784348771
- https://github.com/SaptarshiSarkar12/Drifty/pull/664#discussion_r1784357692

Once you are done, mark those as resolved and let me know. I would approve the changes 😁.

**Review by SaptarshiSarkar12 — APPROVED** [review]
@Ayushmaanagarwal1211 **Looks good to merge 👍.** 
**Thanks for contributing 🚀 🚀.**
You may join our _Discord server_ - **https://discord.gg/DeT4jXPfkG** to get updates about the project.

**Comment by Ayushmaanagarwal1211** [comment]
@SaptarshiSarkar12 thanks for merging

**Comment by SaptarshiSarkar12** [comment]
@Ayushmaanagarwal1211 Most welcome, Ayushman 😁.
Looking forward for more contributions from you 😄.

**Pull request merged**